diff --git a/dlls/ntdll/unix/thread.c b/dlls/ntdll/unix/thread.c
index c3ab1317295..070e93d9649 100644
--- a/dlls/ntdll/unix/thread.c
+++ b/dlls/ntdll/unix/thread.c
@@ -1575,15 +1575,21 @@ NTSTATUS WINAPI NtRaiseException( EXCEPTION_RECORD *rec, CONTEXT *context, BOOL
     if (first_chance) return call_user_exception_dispatcher( rec, context );
 
     if (rec->ExceptionFlags & EH_STACK_INVALID)
-        ERR_(seh)("Exception frame is not in stack limits => unable to dispatch exception.\n");
+        WARN_(seh)("Exception frame is not in stack limits => attempting to continue.\n");
     else if (rec->ExceptionCode == STATUS_NONCONTINUABLE_EXCEPTION)
         ERR_(seh)("Process attempted to continue execution after noncontinuable exception.\n");
     else
-        ERR_(seh)("Unhandled exception code %x flags %x addr %p\n",
+        WARN_(seh)("Unhandled exception code %x flags %x addr %p\n",
                   (int)rec->ExceptionCode, (int)rec->ExceptionFlags, rec->ExceptionAddress );
 
-    NtTerminateProcess( NtCurrentProcess(), rec->ExceptionCode );
-    return STATUS_SUCCESS;
+    if (rec->ExceptionCode == STATUS_NONCONTINUABLE_EXCEPTION)
+    {
+        NtTerminateProcess(NtCurrentProcess(), rec->ExceptionCode);
+        return STATUS_SUCCESS;
+    }
+
+// Attempt to continue execution, even if the exception was not handled
+    return NtContinue(context, FALSE);
 }
 
